### PR TITLE
NF-622 - Protect upstream submission from double quote rejection

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/models/eis/QuoteFormatter.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+}
+
+class QuoteFormatterImpl extends QuoteFormatter {
+
+  override def format(value: String): String = illegalRegex.findFirstIn(value).fold(value)(_ => s"[$value]")
+}
+
+object QuoteFormatterImpl {
+  val illegalRegex: Regex = """^\s*[\"'`;]|[\"'`]\s*$""".r
```
Comment (rorticus0502) [inline app/models/eis/QuoteFormatter.scala:37]: I have tried to find a way past this regex. I cannot. It works

File: app/models/eis/QuoteFormatter.scala [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@ImplementedBy(classOf[QuoteFormatterImpl])
+trait QuoteFormatter {
+
+  def format(value: String): String
+
+}
+
+class QuoteFormatterImpl extends QuoteFormatter {
+
+  override def format(value: String): String = illegalRegex.findFirstIn(value).fold(value)(_ => s"[$value]")
```
Comment (rorticus0502) [inline app/models/eis/QuoteFormatter.scala:33]: This is neat

File: test/models/eis/QuoteFormatterSpec.scala [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.eis
+
+import base.SpecBase
+import org.scalatest.MustMatchers
+
+class QuoteFormatterSpec extends SpecBase with MustMatchers {
```
Comment (rorticus0502) [inline test/models/eis/QuoteFormatterSpec.scala:22]: I love this test.

Please add a couple of tests with leading and trailing white space, as that is part of the regex.

I know they will pass, but may help future dev

**Review by rorticus0502 — CHANGES_REQUESTED** [review]

File: test/models/eis/QuoteFormatterSpec.scala (hunk 3 above)
Comment (SteveSugden) [reply test/models/eis/QuoteFormatterSpec.scala:22]: Good shout.   I hadn't spotted that in the regex - it's not explicitly called out in the ticket.    So the "formatting" should kick in for space followed by illegal character at the beginning or illegal character followed by space at the end.   Agreed?

**Review by rorticus0502 — DISMISSED** [review]

File: test/models/eis/QuoteFormatterSpec.scala (hunk 3 above)
Comment (rorticus0502) [reply test/models/eis/QuoteFormatterSpec.scala:22]: Agree exactly.

**Review by rorticus0502 — APPROVED** [review]

**Pull request merged**